### PR TITLE
fix(admin): fix potentially truncate arguments

### PR DESCRIPTION
**Pull request by vm-001**
### Summary

Admin API will quietly truncate the request post arguments if reach the limit(100), which might cause data inconsistency issues.

### Full changelog

1. Increase max_uri_args and max_post_args to 1000
2. Return 400 while reach the maximum argument number

### Issue reference

[FTI-4407]

[FTI-4407]: https://konghq.atlassian.net/browse/FTI-4407?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: kong/api/arguments.lua [hunk 1]
```diff
@@ -631,7 +632,12 @@ local function load(opts)
     post = {},
   }, arguments_mt)
 
-  local uargs = get_uri_args(options.max_uri_args)
+  local uargs, err
+  uargs, err = get_uri_args(options.max_uri_args)
```
Comment (chronolaw) [inline kong/api/arguments.lua:636]: We can still use `local uargs, err = ...`, I think it is more readable.

File: kong/api/arguments.lua [hunk 2]
```diff
@@ -649,7 +655,11 @@ local function load(opts)
 
   if find(content_type_lower, "application/x-www-form-urlencoded", 1, true) == 1 then
     req_read_body()
-    local pargs, err = get_post_args(options.max_post_args)
+    local pargs
+    pargs, err = get_post_args(options.max_post_args)
```
Comment (chronolaw) [inline kong/api/arguments.lua:659]: Same as above comment.

File: spec/02-integration/04-admin_api/21-truncated_arguments_spec.lua [hunk 3]
```diff
@@ -0,0 +1,70 @@
+local helpers = require "spec.helpers"
+
+for _, strategy in helpers.each_strategy() do
+
+  describe("Admin API - truncated arguments #" .. strategy, function()
+    local max_uri_args = 1000
```
Comment (chronolaw) [inline spec/02-integration/04-admin_api/21-truncated_arguments_spec.lua:6]: How about `local max_uri_args_overflow = 1000 + 1`

File: kong/api/arguments.lua [hunk 4]
```diff
@@ -631,7 +632,12 @@ local function load(opts)
     post = {},
   }, arguments_mt)
 
-  local uargs = get_uri_args(options.max_uri_args)
+  local uargs, err
+  uargs, err = get_uri_args(options.max_uri_args)
+  if err == "truncated" then
+    kong.log.err("Too many arguments")
```
Comment (mayocream) [inline kong/api/arguments.lua:638]: Seems unnecessary to me.

Comment (vm-001) [reply kong/api/arguments.lua:638]: Ops! This is my debug code.

File: spec/02-integration/04-admin_api/21-truncated_arguments_spec.lua [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      client = helpers.admin_client()
+    end)
+
+    after_each(function()
+      if client then
+        client:close()
+      end
+    end)
+
+    it("return 400 if reach maxinum post arguments number", function()
```
Comment (chronolaw) [inline spec/02-integration/04-admin_api/21-truncated_arguments_spec.lua:34]: typo, shoud be `maximum`

File: spec/02-integration/04-admin_api/21-truncated_arguments_spec.lua [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+          ["Content-Type"] =  "application/x-www-form-urlencoded",
+        },
+        body = body,
+      })
+      assert.response(res).has.status(400)
+      local json = assert.response(res).has.jsonbody()
+      assert.same({ message = "Too many arguments" }, json)
+    end)
+
+    it("return 400 if reach maxinum uri arguments number", function()
```
Comment (chronolaw) [inline spec/02-integration/04-admin_api/21-truncated_arguments_spec.lua:53]: typo, should be `maximum`

**Review by fffonion — APPROVED** [review]

**Comment by chronolaw** [comment]
Should we add an entry in changelog?

**Comment by fffonion** [comment]
Yes, right thanks for reminding @chronolaw ! 
@vm-001 Could you create another PR to add changelog?

**Pull request merged**